src/clib/pioc.c

```diff
@@ -2085,14 +2085,14 @@ PIOc_init_async_from_comms(MPI_Comm world, int component_count, MPI_Comm *comp_c
         return check_mpi(NULL, NULL, ret, __FILE__, __LINE__);
 
     /* Get num_procs_per_comp for each comp and share with world */
-    num_procs_per_comp = (int *) calloc(component_count, sizeof(int));
+    num_procs_per_comp = (int *) malloc(component_count * sizeof(int));
```

Comment:
You are changing these so they are no longer pre-filled with zeros. Is that intentional?

Reply:
I am 0 filling them below - calloc seemed to be giving an error in address sanitizer.   Currently the test
pio__iosystem_async_tests passes with mpich and mpt but fails with openmpi.

Reply:
OK, you want to mention that in the comments somewhere...

Reply:
I'm not really sure that was the problem.  Now that I have it working I could try putting back the callocs.

Reply:
OK, if you want to just leave it, it's not a big deal...

Generally if I see mallocs + setting to zero I will automatically refactor to callocs whenever I tough the code anyway. ;-)